src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -1645,6 +1697,23 @@ void MacroAssembler::movptr(Register Rd, address addr, int32_t &offset) {
   offset = imm64 & 0x3f;
 }
 
+void MacroAssembler::movptr2(Register Rd, uint64_t addr, int32_t &offset, Register tmp) {
+  assert_different_registers(Rd, tmp, noreg);
+
+  uint32_t upper18 = (addr >> 30ull);
+  int32_t  lower30 = (addr & 0x3fffffffu);
+  int32_t  low12   = (lower30 << 20) >> 20;
+  int32_t  mid18   = ((lower30 - low12) >> 12);
```

Comment:
Similar here. `mid18 = ((lower30 - low12) >> 12);` and `lui(Rd, mid18 << 12);` could be replaced with `lui(Rd, lower30 - low12);`?

Reply:
Same reason as above.